pyanaconda/modules/payload/sources/live_os.py

```diff
@@ -0,0 +1,137 @@
+from pyanaconda.core.util import execWithCapture
+from pyanaconda.modules.payload.base.constants import SourceType
+from pyanaconda.modules.payload.base.source_base import PayloadSourceBase
+from pyanaconda.modules.payload.sources.live_os_interface import LiveOSSourceInterface
+from pyanaconda.modules.payload.sources.initialization import SetUpInstallationSourceTask, \
+    TearDownInstallationSourceTask
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class LiveOSSourceModule(PayloadSourceBase):
+    """The Live OS source payload module."""
+
+    def __init__(self):
+        super().__init__()
+        self._image_path = ""
+        self.image_path_changed = Signal()
+
+    @property
+    def kind(self):
+        """Get type of this source."""
+        return SourceType.LIVE_OS_IMAGE
+
+    @property
+    def image_path(self):
+        """Path to the source live OS image.
+
+        This image will be used for the installation.
+
+        :rtype: str
+        """
+        return self._image_path
+
+    def set_image_path(self, image_path):
+        """Set path to the live os OS image.
+
+        :param image_path: path to the image
+        :type image_path: str
+        """
+        self._image_path = image_path
+        self.image_path_changed.emit()
+        log.debug("LiveOS image path is set to '%s'", self._image_path)
+
+    def for_publication(self):
+        """Get the interface used to publish this source."""
+        return LiveOSSourceInterface(self)
+
+    def detect_live_os_base_image(self):
+        """Detect live os image in the system."""
+        log.debug("Trying to detect live os base image automatically")
+        for block_device in ["/dev/mapper/live-base", "/dev/mapper/live-osimg-min"]:
+            try:
+                if stat.S_ISBLK(os.stat(block_device)[stat.ST_MODE]):
+                    log.debug("Detected live base image %s", block_device)
+                    return block_device
+            except FileNotFoundError:
+                pass
+
+        # Is it a squashfs+overlayfs base image?
+        if os.path.exists("/run/rootfsbase"):
+            try:
+                block_device = execWithCapture("findmnt",
+                                               ["-n", "-o", "SOURCE", "/run/rootfsbase"]).strip()
+                if block_device:
+                    log.debug("Detected live base image %s", block_device)
+                    return block_device
+            except (OSError, FileNotFoundError):
+                pass
+
+        log.debug("No live base image detected")
+        return ""
+
+    def set_up_with_tasks(self):
+        """Set up the installation source for installation.
+
+        :return: list of tasks required for the source setup
+        :rtype: [Task]
+        """
+        task = SetUpInstallationSourceTask(self._image_path, INSTALL_TREE)
+
+        task.succeeded_signal.connect(lambda: self._set_is_ready(True))
```

Comment:
I am afraid that the `is_ready` property will cause us a lot of pain in the future. Since the payload handler runs these tasks, couldn't the payload handler track the readiness of its sources?

Reply:
Btw. in this case `is_ready` just means that the source is mounted at the expected path.

Reply:
Yes this is just a dumb solution in general the `is_ready` implementation should check the state on the disk.

Reply:
I think this should be enough for the Live Source but I want to change this behavior in further implementation.